src/buffer/user_context.rs

```diff
@@ -80,42 +84,103 @@ pub fn view<'a>(
 
         let (content, message) = match entry {
-            Entry::Whois => ("Whois", Message::Whois(nickname)),
-            Entry::Query => ("Message", Message::Query(nickname)),
+            Entry::Whois => (button_text("Whois"), Some(Message::Whois(nickname))),
+            Entry::Query => (button_text("Message"), Some(Message::Query(nickname))),
             Entry::ToggleAccessLevelOp => {
                 if user.has_access_level(data::user::AccessLevel::Oper) {
                     (
-                        "Take Op (-o)",
-                        Message::ToggleAccessLevel(nickname, "-o".to_owned()),
+                        button_text("Take Op (-o)"),
+                        Some(Message::ToggleAccessLevel(nickname, "-o".to_owned())),
                     )
                 } else {
                     (
-                        "Give Op (+o)",
-                        Message::ToggleAccessLevel(nickname, "+o".to_owned()),
+                        button_text("Give Op (+o)"),
+                        Some(Message::ToggleAccessLevel(nickname, "+o".to_owned())),
                     )
                 }
             }
             Entry::ToggleAccessLevelVoice => {
                 if user.has_access_level(data::user::AccessLevel::Voice) {
                     (
-                        "Take Voice (-v)",
-                        Message::ToggleAccessLevel(nickname, "-v".to_owned()),
+                        button_text("Take Voice (-v)"),
+                        Some(Message::ToggleAccessLevel(nickname, "-v".to_owned())),
                     )
                 } else {
                     (
-                        "Give Voice (+v)",
-                        Message::ToggleAccessLevel(nickname, "+v".to_owned()),
+                        button_text("Give Voice (+v)"),
+                        Some(Message::ToggleAccessLevel(nickname, "+v".to_owned())),
                     )
                 }
             }
-            Entry::SendFile => ("Send File", Message::SendFile(nickname)),
+            Entry::SendFile => (button_text("Send File"), Some(Message::SendFile(nickname))),
+            Entry::UserInfo => (user_info(current_user, length), None),
         };
 
-        button(text(content).style(theme::text::primary))
-            .padding(5)
-            .width(length)
-            .style(theme::button::context)
-            .on_press(message)
-            .into()
+        if let Some(message) = message {
+            button(content)
+                .padding(5)
+                .width(length)
+                .style(theme::button::context)
+                .on_press(message)
+                .into()
+        } else {
+            column![]
+                .push(container(content).padding(5).width(length))
+                .push(
+                    row![]
+                        .push(horizontal_rule(1))
+                        .padding([0, 5])
+                        .width(length),
```

Comment:
Does this have to be a `row![]`?

Reply:
I made it a `row![]` to be able to apply the `padding` and `width` to it.  I know how to move the `padding` to the other widgets, but I don't know how to get the effects of `width` without the `row![]` (without `width` the `horizontal_rule` makes the context menu very wide).